packages/arb-util/broadcastclient/broadcastclient.go

```diff
@@ -53,31 +61,75 @@ type BroadcastClient struct {
 
 var logger = arblog.Logger.With().Str("component", "broadcaster").Logger()
 
-func NewBroadcastClient(websocketUrl string, lastInboxSeqNum *big.Int, idleTimeout time.Duration) *BroadcastClient {
-	var seqNum *big.Int
-	if lastInboxSeqNum == nil {
-		seqNum = big.NewInt(0)
+func NewBroadcastClient(
+	websocketUrl string,
+	chainId uint64,
+	requestedInboxSeqNum *big.Int,
+	idleTimeout time.Duration,
+	broadcastClientErrChan chan error,
+) *BroadcastClient {
+	var lastSeqNum *big.Int
+	if requestedInboxSeqNum == nil || requestedInboxSeqNum.Cmp(big.NewInt(0)) <= 0 {
+		lastSeqNum = big.NewInt(0)
 	} else {
-		seqNum = lastInboxSeqNum
+		lastSeqNum = new(big.Int).Sub(requestedInboxSeqNum, big.NewInt(1))
 	}
 
 	return &BroadcastClient{
 		websocketUrl:    websocketUrl,
-		lastInboxSeqNum: seqNum,
+		chainId:         chainId,
+		lastInboxSeqNum: lastSeqNum,
 		connMutex:       &sync.Mutex{},
+		errChan:         broadcastClientErrChan,
 		retryMutex:      &sync.Mutex{},
 		idleTimeout:     idleTimeout,
 	}
 }
 
+func (bc *BroadcastClient) ChainId() uint64 {
+	return atomic.LoadUint64(&bc.chainId)
+}
+
+func (bc *BroadcastClient) SetChainId(newChainId uint64) error {
```

Comment:
Looks like this is used just for the relay. I think it'd be a lot simpler to require the relay get passed a l2 chain id when booted and then just exit if it receives a mismatching one. Then I think we can drop all the atomicity since we won't need to mutate after construction

Reply:
Requiring `--l2.chain-id` for arb-relay allows for a bunch of code simplification